packages/artifact/src/internal-artifact-client.ts

```diff
@@ -100,25 +133,114 @@ export class DefaultArtifactClient implements ArtifactClient {
     return uploadResponse
   }
 
-  /*
-  Downloads a single artifact associated with a run
+  async downloadArtifact(
+    name: string,
+    path?: string | undefined,
+    options?: DownloadOptions | undefined
+  ): Promise<DownloadResponse> {
+    const artifacts = await listArtifacts()
+    if (artifacts.count === 0) {
+      throw new Error(
+        `Unable to find any artifacts for the associated workflow`
+      )
+    }
+
+    const artifactToDownload = artifacts.value.find(artifact => {
+      return artifact.name === name
+    })
+
+    if (!artifactToDownload) {
+      throw new Error(`Unable to find an artifact with the name: ${name}`)
+    } else {
+      // query container for all items associated with the artifact
+      const items = await getContainerItemsForSingleArtifact(
+        artifactToDownload.name,
+        artifactToDownload.fileContainerResourceUrl
+      )
+
+      if (!path) {
+        path = getWorkSpaceDirectory()
+      }
 
-  export async function downloadArtifact(
-      name: string,
-      path?: string,
-      options?: DownloadOptions
-    ): Promise<DownloadResponse> {
+      // Check only for downloadable items that are not directories. During upload, empty directories are rejected by the remote server so there should be no
+      // artifacts that consist of only empty directories
+      if (items.value.length === 0) {
+        throw new Error(
```

Comment:
I don't think we should throw here.  If the artifact is empty it is empty.  We can log an info here saying no items were found.  

The above will happen if a file is uploaded partially and never finished.

Reply:
Changed it to just `core.info`. The returned `DownloadResponse` will be the same so it will be up to the user to figure out that the artifact is actually empty if they don't look at the logging somehow